### ISSUE TITLE
Shader updates not working for reused shaders

### ISSUE THREAD

**Issue by emiax** [OpenSpace/OpenSpace#818]
The following pattern, used throughout the code is problematic since shaders may be shared across renderables.

``` 
if (_program->isDirty()) {
    _program->rebuildFromFile();
    ghoul::opengl::updateUniformLocations(*_program, _uniformCache, UniformNames);
}
```

Only the first renderable updates its cached uniform locations, while others don't get the chance to detect that the shader was dirty and rebuilt.

This is related to #525 (Improve shader reuse)

**Comment by alexanderbock** [comment]
Moved this to 0.15.0 as it requires a bigger change that might be iffy to do shortly before the release

**Comment by alexanderbock** [comment]
Closing it for now until we implemented #525 and we need this solution again
